packages/processors/src/helpers/pricing.ts

```diff
@@ -0,0 +1,71 @@
+import { IPricingProvider } from "@grants-stack-indexer/pricing";
+import { Token } from "@grants-stack-indexer/shared";
+
+import { TokenPriceNotFoundError } from "../internal.js";
+import { calculateAmountInToken, calculateAmountInUsd } from "./tokenMath.js";
+
+// sometimes coingecko returns no prices for 1 hour range, 2 hours works better
+const TIMESTAMP_DELTA_RANGE = 2 * 60 * 60 * 1000;
+
+/**
+ * Get the amount in USD for a given amount in the token
+ * @param pricingProvider - The pricing provider to use
+ * @param token - The token to get the amount in
+ * @param amount - The amount in the token
+ * @param timestamp - The timestamp to get the price at
+ * @returns The amount in USD
+ * @throws TokenPriceNotFoundError if the price is not found
+ */
+export const getTokenAmountInUsd = async (
+    pricingProvider: IPricingProvider,
+    token: Token,
+    amount: bigint,
+    timestamp: number,
+): Promise<{ amountInUsd: string; timestamp: number }> => {
+    const tokenPrice = await pricingProvider.getTokenPrice(
+        token.priceSourceCode,
+        timestamp,
+        timestamp + TIMESTAMP_DELTA_RANGE,
```

Comment:
in addition to prev comment, instead timestamp + TIMESTAMP_DELTA_RANGE as param , just delta